### PR TITLE
ACT: add quickfix to create lifetime parameter from usage

### PR DESCRIPTION
**Pull request by sirgl**
Related issue: #3183

### CODE REVIEW

**Comment by vlad20012** [comment]
Hah, your gif example shows incorrect behavior =)
Lifetimes inside inner structs are independent from parent declaration, so this example will not be compiled
```rust
fn foo<'a>() {
    struct S { f: &'a i32 }
}
```

**Comment by vlad20012** [comment]
(and this is also a bug in our lifetime resolution ):

**Comment by sirgl** [comment]
That's happens when you trust IDE too much :)

File: src/main/kotlin/org/rust/ide/annotator/fixes/CreateLifetimeParameterFromUsageFix.kt [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+    }
+
+    override fun getText(): String = familyName
+
+    override fun invoke(project: Project, file: PsiFile, editor: Editor?, startElement: PsiElement, endElement: PsiElement) {
+        val context = gatherContext(startElement) ?: return
+        insertLifetime(context.declaration, project, context.lifetime)
+    }
+
+    private fun insertLifetime(declaration: RsGenericDeclaration, project: Project, startElement: RsLifetime) {
```
Comment (vlad20012) [inline src/main/kotlin/org/rust/ide/annotator/fixes/CreateLifetimeParameterFromUsageFix.kt]: Looks like `startElement` is a temporal name and should be renamed to `lifetime` or something

**Comment by vlad20012** [comment]
Some tests are failed. Looks like it's related to recent changes in annotator tests #3159

**Comment by Undin** [comment]
@sirgl Thanks!
I've fixed tests and merged this feature in #3385

**Pull request closed without merging**